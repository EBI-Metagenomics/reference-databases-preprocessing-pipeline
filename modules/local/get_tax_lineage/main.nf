
process GET_TAX_LINEAGE {

    label 'process_low'

    input:
    path fasta
    path taxdump
    val version
    val label

    output:
    path("*tax_lineage.txt"), emit: tax_lineage

    script:
    """
    grep ">" $fasta | cut -d"|" -f2 > itsonedb.sliced_taxids
    unzip $taxdump 
    get_tax_lineage.py -t itsonedb.sliced_taxids -d taxa.txt -o itsonedb_tax_lineage.txt
    """

}