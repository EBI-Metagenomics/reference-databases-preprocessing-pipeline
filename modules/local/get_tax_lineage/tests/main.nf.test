nextflow_process {

    name "Test Process GET_TAX_LINEAGE"
    script "modules/local/get_tax_lineage/main.nf"
    process "GET_TAX_LINEAGE"
    tag "modules"
    tag "get_tax_lineage"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = file('${projectDir}/modules/local/get_tax_lineage/tests/data/test.fasta', checkIfExists: true)
                input[1] = file('${projectDir}/modules/local/get_tax_lineage/tests/data/taxa.zip', checkIfExists: true)
                input[2] = params.itsonedb_version
                input[3] = params.itsonedb_label
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tax_lineage).match("tax_lineage") }
            )
        }

    }

}
