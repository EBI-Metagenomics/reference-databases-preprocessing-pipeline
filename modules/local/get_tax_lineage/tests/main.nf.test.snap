{
    "tax_lineage": {
        "content": [
            [
                "itsonedb_tax_lineage.txt:md5,d0d91ad37327ea0ed0bd1da9b3735013"
            ]
        ],
        "timestamp": "2023-12-12T15:32:08.204002307"
    }
}